src/devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,319 @@
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Bmp180
+{
+    /// <summary>
+    /// Bmm150 class implementing a magnetometer
+    /// </summary>
+    [Interface("Bmm150 class implementing a magnetometer")]
+    public sealed class Bmm150 : IDisposable
+    {
+        /// <summary>
+        /// I2c device comm channel
+        /// </summary>
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Bmm150 device interface
+        /// </summary>
+        private Bmm150I2cBase _bmm150Interface;
+
+        /// <summary>
+        /// Bmm150 Trim extended register data
+        /// </summary>
+        private Bmm150TrimRegisterData _trimData;
+
+        /// <summary>
+        /// Magnetometer (R-HALL) temperature compensation value, used in axis compensation calculation functions
+        /// </summary>
+        private uint _rHall;
+
+        /// <summary>
+        /// Flag to evaluate disposal of resources
+        /// </summary>
+        private bool _shouldDispose = true;
+
+        /// <summary>
+        /// Gets or sets Magnetometer calibration compensation vector
+        /// </summary>
+        public Vector3 CalibrationCompensation { get; set; } = new Vector3();
+
+        /// <summary>
+        /// Primary I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L163
+        /// </summary>
+        public const byte PrimaryI2cAddress = 0x13;
+
+        /// <summary>
+        /// Secondary I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13, alhtough for m5stack is 0x10
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x10;
+
+        /// <summary>
+        /// Default timeout to use when timeout is not provided in the reading methods
+        /// </summary>
+        [Property]
+        public TimeSpan DefaultTimeout { get; set; } = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Default constructor for an independent Bmm150
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device</param>
+        public Bmm150(I2cDevice i2CDevice)
+            : this(i2CDevice, new Bmm150I2c())
+        {
+        }
+
+        /// <summary>
+        /// Constructor to use if Bmm150 is behind another element and need a special I2C protocol like
+        /// when used with the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="Bmm150Interface">The specific interface to communicate with the Bmm150</param>
+        /// <param name="shouldDispose">True to dispose the I2C device when class is disposed</param>
+        public Bmm150(I2cDevice i2cDevice, Bmm150I2cBase Bmm150Interface, bool shouldDispose = true)

```

Comment:
I don't think we need the "shouldDispose" here. The `I2cDevice` instance is always bound to a binding and shall not be shared.

Reply:
That is needed if you are using an I2C interface that is built over a regular I2C Device in the case of BMI160 for example.